spec.emu

```diff
@@ -0,0 +1,108 @@
+<pre class="metadata">
+title: JavaScript Module Blocks
+stage: 1
+contributors: Surma, Dan Ehrenberg
+</pre>
+
+<emu-clause id="sec-primary-expression">
+  <h1>Primary Expression</h1>
+  <h2>Syntax</h2>
+  <emu-grammar type="definition">
+    PrimaryExpression[Yield, Await] :
+      IdentifierReference[?Yield, ?Await]
+      Literal
+      ArrayLiteral[?Yield, ?Await]
+      ObjectLiteral[?Yield, ?Await]
+      FunctionExpression
+      ClassExpression[?Yield, ?Await]
+      GeneratorExpression
+      AsyncFunctionExpression
+      AsyncGeneratorExpression
+      <ins>ModuleBlockExpression</ins>
+      RegularExpressionLiteral
+      TemplateLiteral[?Yield, ?Await, ~Tagged]
+      CoverParenthesizedExpressionAndArrowParameterList[?Yield, ?Await] #parencover
+  </emu-grammar>
+</emu-clause>
+
+<emu-clause id="sec-module-block-definitions">
+  <h1>Module Block Definitions</h1>
+  <h2>Syntax</h2>
+  <emu-grammar type="definition">
+    ModuleBlockExpression :
+      `module` [no LineTerminator here] `{` ModuleBody? `}`
+  </emu-grammar>
+
+  <h2>Runtime Semantics: Evaluation</h2>
+  <emu-alg>
+    1. Let _moduleBlock_ be OrdinaryObjectCreate(%ModuleBlock.prototype%,  &laquo; [[HostDefinedSlot]], [[ModuleBlockBody]], [[SourceText]] &raquo;).
+    1. Set _moduleBlock_.[[SourceText]] to the source text matched by |ModuleBlockExpression|.
+    1. Perform ! HostInitializeModuleBlock(_moduleBlock_).
+    1. Return _moduleBlock_.
+  </emu-alg>
+</emu-clause>
+  
+<emu-clause id="sec-moduleblock">
+  <h1>ModuleBlock ( _body_ )</h1>
+  <p>The _body_ specifies the body (executable code) of a module block.</p>
+  <p>When the `ModuleBlock` function is called,the following steps are taken:</p>
+  <emu-alg>
+    1. If NewTarget is *undefined*
+      1. Throw a *TypeError* exception.
+    1. Let _parsedBody_ be ? ParseText(_body_, Module).
```

Comment:
`_body_` should be a ParseNode, not a text

Reply:
This algorithm is about converting a string into a module block. It looks like ParseText actually accepts a List of code points, so we can't pass in a string directly and should fix this. But I am confused by this comment--why do you say Body should be a Parse Node?

Reply:
I think this is strange. The current draft seems like converting a module block into a string then call the constructor to parse the string

Reply:
This path is only called if you call the ModuleBlock constructor explicitly. It's kinda like the function constructor. The syntax does not go through this algorithm.

Reply:
I didn't see the set to `[[ModuleBlockBody]]` in the RS of the syntax. And why we need a constructor for the module block?